src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -231,6 +232,19 @@ else if (shouldStartLine && !isOnStartOfLine(details, targetSourceLine)) {
     }
 
     /**
+     * Checks that a right curly should be alone on line or alone with SEMI token is LAMBDA.
+     * @param bracePolicy option for placing the right curly brace
+     * @param details Details for validation
+     * @return true if a right curly should be alone on line or with semi if token is LAMBDA.
+     */
+    private static boolean shouldBeAlone(RightCurlyOption bracePolicy, Details details) {
+        return shouldBeAloneOnLine(bracePolicy, details)
+                && details.lcurly.getParent().getType() != TokenTypes.LAMBDA
+                || shouldBeAloneWithSemi(bracePolicy, details)
+                && details.lcurly.getParent().getType() == TokenTypes.LAMBDA;
```

Comment:
1) please move calculation of "details.lcurly.getParent().getType()" to local variable, its calculation is unavoidable.
2) in "&&" conditions please put the quickest operation in front.
3) please intend code of "a && b || c && d" as
```
a
   && b
||  c 
   && d
```
to clear see priority of executions.